### ISSUE TITLE
Use `npm shrinkwrap` to lock down npm versions

### ISSUE THREAD

**Issue by tristil** [neagle/gokibitz#58]
I mentioned this here: https://github.com/neagle/gokibitz/pull/57 but creating a separate issue to track the discussion.

Basically NPM makes two strange decisions when it comes to specifying dependencies for a project:
1) When you do` npm install blah --save` it uses the `^` symbol to specify that the minor number can change. So if you have `^3.0.2` it will update to `3.1`, `3.2` etcetera. This means your end users can have very different configurations from your own install.
2) Even if you hardcode a version number, say `3.0.2`, the dependencies inside that package can vary from install to install.

To be fair these make more sense in terms of having loose dependencies for packages but they are hell for providing a deterministic project.

I'm kind of just ranting for my own benefit because these are things I just figured out for work. Anyway here is some information on it:

https://docs.npmjs.com/cli/shrinkwrap
http://blog.nodejs.org/2012/02/27/managing-node-js-dependencies-with-shrinkwrap/

**Comment by neagle** [comment]
To be fair to the Node community, I think `shrinkwrap` is something experienced project developers know about and use. I still have a lot to learn about managing larger projects, and had ignored the mentions I'd heard of `shrinkwrap` until the place or time that I had to learn a bit more about it, which it turns out was today.

As I noted in my commit message, you also have to include a `--dev` flag if you want your devDependencies to be included, since the idea is to preserve the flexibility in normal semver for developers, but lock down production. One doesn't _have_ to use devDependencies at all, but I've been thinking of the division this way: `dependencies` is for packages I need for actually running and serving up GoKibitz on the server, `devDepencies` is for everything required to develop and run things locally.

But since part of the idea of adding this (from my perspective) is to prevent people like you and @levelonedev from having to do irritating debugging when you're contributing out of the goodness of your hearts, I just ran `npm shrinkwrap --dev` to include devDependencies. Luckily, the default command outputs a bunch of warnings if you have devDependencies in your package.json.

Anyway, thanks! This was good for me to learn about.
